packages/squiggle-lang/src/rescript/ReducerInterface/ReducerInterface_GenericDistribution.res

```diff
@@ -388,6 +387,20 @@ let genericOutputToReducerValue = (o: DistributionOperation.outputType): result<
   | GenDistError(err) => Error(REDistributionError(err))
   }
 
-let dispatch = (call, environment) => {
-  dispatchToGenericOutput(call, environment)->E.O2.fmap(genericOutputToReducerValue)
+let registered = FunctionRegistry.allFunctions
+
+let tryRegistry = (call: ExpressionValue.functionCall) => {
+  let (fnName, args) = call
+  let response = FunctionRegistry.Registry.matchAndRun(registered, fnName, args)
+  let foo = response->E.O2.fmap(r => r->E.R2.errMap(s => Reducer_ErrorValue.RETodo(s)))
+  foo
```

Comment:
don't forget to shorten this to just returning the `->` expression and killing the name `foo`